tests/test_unit.py

```diff
@@ -667,67 +668,69 @@ def test_parse_markdown_header_alternate(self):
 
         self.assertEqual(header_line_got, header_line_want)
 
-        # The header should be the same even with the license header.
-        header_line_with_license_want = header_line_want
-
-        with open('tests/markdown_example_alternate_header.md', 'r') as mdfile:
-            header_line_with_license_got = extract_header_from_markdown(mdfile)
-
-        self.assertEqual(header_line_with_license_got, header_line_with_license_want)
 
-        # Check the header for an alternate header style.
+    test_markdown_filenames = [
+        [
+            # Check the header for an alternate header style.
+            "tests/markdown_example_alternate.md"
+        ],
+        [
+            # The header should be the same even with the license header.
+            "tests/markdown_example_alternate_header.md"
+        ],
+        [
+            # Check the header for an alternate header style.
+            "tests/markdown_example_alternate_less.md"
+        ],
+    ]
+    @parameterized.expand(test_markdown_filenames)
+    def test_extract_header_from_markdown_alternate_header(self, markdown_filename):
+        # Check the header for different accepted styles.
         header_line_want = "This is a simple alternate header"
 
-        with open('tests/markdown_example_alternate_less.md', 'r') as mdfile:
+        with open(markdown_filename, 'r') as mdfile:
             header_line_got = extract_header_from_markdown(mdfile)
 
         self.assertEqual(header_line_got, header_line_want)
 
 
-    def test_extract_header_from_markdown_bad_headers(self):
-        # Check that the filename is used as header if no valid header is found.
+    test_markdown_filenames = [
+        [
+            "tests/markdown_example_bad_header.md"
+        ],
+        [
+            "tests/markdown_example_h2.md"
+        ],
+        [
+            "tests/markdown_example_alternate_bad.md"
+        ],
+    ]
+    @parameterized.expand(test_markdown_filenames)
+    def test_extract_header_from_markdown_bad_headers(self, markdown_filename):
+        # Check that empty string is returned if no valid header is found.
         header_line_want = ""
 
-        with open('tests/markdown_example_bad_header.md', 'r') as mdfile:
-            header_line_got = extract_header_from_markdown(mdfile)
-
-        self.assertEqual(header_line_want, header_line_got)
-
-        # Check that only h1 headers are parsed.
-        with open('tests/markdown_example_h2.md', 'r') as mdfile:
-            header_line_got = extract_header_from_markdown(mdfile)
-
-        self.assertEqual(header_line_want, header_line_got)
-
-        # Check that there must be a line before the h1 header breaker.
-        with open('tests/markdown_example_alternate_bad.md', 'r') as mdfile:
+        with open(markdown_filename, 'r') as mdfile:
             header_line_got = extract_header_from_markdown(mdfile)
 
         self.assertEqual(header_line_want, header_line_got)
```

Comment:
assertEmpty

Reply:
I thought assertEmpty exists but assertTrue/assertFalse is used instead apparently 😮